### PR TITLE
Update only the least-recently updated PR

### PR DESCRIPTION
**Pull request by josephfrazier**
This is a first attempt at implementing a "merge queue" as outlined in https://github.com/tibdex/auto-update/issues/10

The logic behind using the least-recently updated PR is that it's the
one that's most likely to be the "most done" of the auto-merge-enabled
PRs, due either to being auto-merge-enabled first, or to there being no
further comments on it.

### CODE REVIEW

**Review by ericgross — APPROVED** [review]
Yeah - I think we would not want to test this on a repo that is important at all.

File: src/index.ts [hunk 1]
```diff
@@ -174,9 +176,12 @@ const run = async () => {
...
 
     for (const pullRequest of pullRequests) {
-      // PRs are handled sequentially to avoid breaking GitHub's log grouping feature.
-      // eslint-disable-next-line no-await-in-loop
-      await handlePullRequest(pullRequest, { eventPayload, octokit });
+      if (pullRequest.auto_merge) {
+        // PRs are handled sequentially to avoid breaking GitHub's log grouping feature.
+        // eslint-disable-next-line no-await-in-loop
+        await handlePullRequest(pullRequest, { eventPayload, octokit });
+        break;
```
Comment (mduleone) [inline src/index.ts:183]: I would lean into an early `return` here rather than a `break` if there's nothing for us to do once we're out, anyway.

Comment (josephfrazier) [reply src/index.ts:183]: great point, will do
```suggestion
        return;
```

**Comment by vasylatCO** [comment]
what about trying that in master branch? lol

**Comment by vasylatCO** [comment]
> what about trying that in master branch? lol

I guess it will not work as the configuration is different now. sorry, I got preemptively excited about a bad idea 🤷

**Comment by josephfrazier** [comment]
> I'm not sure how to test this, other than publishing the action to github and trying it out in a repo (maybe we make a new one specifically for testing against?)

So it looks like this repo uses itself, see here for example: https://github.com/careofvitamins/auto-update/runs/5160300310?check_suite_focus=true

This means that we should be able to test it by opening a PR that's not up to date with main and doesn't pass the tests, enable auto-merge on it, then make it pass the tests, I think.

**Pull request merged**